### ISSUE TITLE
Transformation breaks on PHP 7.x

### ISSUE THREAD

**Issue by eternoendless** [cssjanus/php-cssjanus#14]
We use this library at PrestaShop to perform RTL transformations onto our CSS.

Recently we became aware that some of our CSS were getting broken by this transformation.

When calling `CSSJanus::transform()` with a sufficiently complex/long CSS ([see a specific offender here](https://github.com/PrestaShop/PrestaShop/blob/1.7.3.x/admin-dev/themes/new-theme/public/theme.css)), we were getting an empty string as return value.

After some investigation, we have found that [this particular regex](https://github.com/cssjanus/php-cssjanus/blob/master/src/CSSJanus.php?utf8=%E2%9C%93#L260) fails, only in PHP >= 7.x, resulting in an emtpy `$css`:
```php
$css = preg_replace(self::$patterns['left'], self::$patterns['tmpToken'], $css);
```

When printing out the last preg error: 
```php
$css = preg_replace(self::$patterns['left'], self::$patterns['tmpToken'], $css);
var_dump(preg_last_error());
```

...we get an error code...
```
int(6)
```
...that matches the constant `PREG_JIT_STACKLIMIT_ERROR` ([see here](http://nz2.php.net/manual/en/pcre.constants.php)).

This points out that the performed regex is overflowing the stack limit of the PCRE JIT compiler found in PHP >= 7.

I don't know if this regex can be improved, it looks pretty complex and obscure to me:
```php
var_dump(self::$patterns['left']);
```
```
string(275) "/(?<![a-zA-Z])(left)(?![a-zA-Z])(?!(?:[!#$%&*-~]|[\200-\377]|(?:(?:(?:\[0-9a-f]{1,6})(?:\r\n|\s)?)|\[^
0-9a-f]))*?['\"]?\s*\))(?!((?:[_a-z0-9-]|[\200-\377]|(?:(?:(?:\[0-9a-f]{1,6})(?:\r\n|\s)?)|\[^
?
|\s|#|\:|\.|\,|\+|>|\(|\)|\[|\]|=|\*=|~=|\^=|'[^']*'])*?{)/i"
```

A simple workaround is simply disabling the PCRE JIT compiler (with the performance impact that involves):

```php
ini_set('pcre.jit', false);
```

I hope this can be fixed eventually 🙂

**Comment by s7eph4n** [comment]
Same problem when transforming the Bootstrap CSS (~300kB of CSS).

**Comment by s7eph4n** [comment]
One more observation: The workaround works. 
What does not work is turning the JIT compiler off and back on again later, e.g. something like
``` php
$jitStatus = ini_set( 'pcre.jit', 0 );

$style = CSSJanus::transform( $style, true, false );

ini_set( 'pcre.jit', $jitStatus );
```

Turning the JIT compiler back on will for some reason trigger the error even though the offending operation is done already.

**Comment by eternoendless** [comment]
Yeah, the workaround that initially seemed to work didn't do the trick in the end. We fixed our problem by forking cssjanus and applying the patch in #13.

**Comment by jdforrester** [comment]
This was fixed by #19.
